code/processes/dqc.q

```diff
@@ -0,0 +1,242 @@
+  `.dqe.results insert (idnum;funct;parprint;rs[0];rs[1];.proc.cp[];0Np;0b;"";`started;runtype);
+  }
+
+updresultstab:{[runtype;idnum;end;res;des;status;params;proc]                                                   /- general function used to update a check in the results table
+  .lg.o[`updresultstab;"Updating check id ",(string idnum)," in the results table with status ",string status];
+  if[1b=params`comp;proc:params`compresproc];
+  if[c:count s:exec i from .dqe.results where id=idnum, procschk=proc,chkstatus=`started;                       /- obtain count of checks that will be updated
+    .lg.o[`updresultstab;raze "run check id ",(string idnum)," update in results table with check status ",string status];
+    `.dqe.results set update endtime:end,result:res,descp:enlist des,chkstatus:status,chkruntype:runtype from .dqe.results where id=idnum,procschk=proc,chkstatus=`started];
+    .dqe.tosavedown[`.dqe.results],:s;
+  delete from `.dqe.compcounter where id=idnum;
+  params:()!();
+  s2:exec i from .dqe.configtable where checkid=idnum;
+  .dqe.tosavedown[`.dqe.configtable]:.dqe.tosavedown[`.dqe.configtable] union s2;
+  }
+
+chkcompare:{[runtype;idnum;params]                                                                              /- function to compare the checks
+  if[params[`compcount]<>(d:.dqe.compcounter idnum)`counter;:()];                                               /- checks if all async check results have returned
+  .lg.o[`chkcompare;"comparison started with id ",string idnum];
+  a:d[`results] where not d[`procs]=params`compproc;                                                            /- obtain all the check returns
+  procsforcomp:d[`procs] except params`compproc;
+  b:d[`results] where d[`procs]=params`compproc;                                                                /- obtain the check to compare the others to
+
+  if[@[{all 0W=x};first b;0b];                                                                                  /- if error in compare proc then fail check
+    .dqe.updresultstab[runtype;idnum;.proc.cp[];0b;"error: error on comparison process";`failed;params;`];:()];
+  errorprocs:d[`procs] where (),all each @[{0W=x};d`results;0b];
+  if[(count errorprocs)= count d`results;                                                                       /- if error in all comparison procs then fail check
+    .dqe.updresultstab[runtype;idnum;.proc.cp[];0b;"error: error with all comparison procs";`failed;params;`];:()];
+  $[@[{98h=type raze x};b;0b];                                                                                  /- changes comparison for tables
+    [matching:procsforcomp where (), params[`compallow] <= (sum t2)%count t2:100*(sum  t)%count t:$[`error~.[{(all/)=[raze x;raze y]};(a;b);{`error}]; 
+      .dqe.updresultstab[runtype;idnum;.proc.cp[];0b;"error: tables are not of the same length";`complete;params;`];
+      :()];
```

Comment:
This entire conditional is quite complicated and convoluted. Is this supposed to be an early exit condition?

Reply:
The conditional was put in there as there was a problem with comparing tables that were different lengths

Reply:
While testing noticed something - would we need to add a conditional for when config table is empty(the current branch that we removed all the configuration)?